kubeflow/trainer/backends/base.py

```diff
@@ -14,13 +14,88 @@
 
 import abc
 from collections.abc import Iterator
+from enum import Enum
 from typing import Optional, Union
 
 from kubeflow.trainer.constants import constants
 from kubeflow.trainer.types import types
 
 
+class OptionType(Enum):
+    """Enumeration of available option types for backends."""
+
+    LABELS = "labels"
+    ANNOTATIONS = "annotations"
+    POD_TEMPLATE_OVERRIDES = "pod_template_overrides"
+    NAME = "name"
+    TRAINER_COMMAND = "trainer_command"
+    TRAINER_ARGS = "trainer_args"
+
+
+class BackendCapabilities:
+    """Backend capabilities for validating option compatibility."""
+
+    def __init__(self, supported_options: set[OptionType]):
+        self.supported_options = supported_options
+        self._supported_types = frozenset(opt.value for opt in supported_options)
+
+    def supports(self, option_type: OptionType) -> bool:
+        """Check if option type is supported."""
+        return option_type.value in self._supported_types
+
+    def supports_option(self, option) -> bool:
+        """Check if option is supported."""
+        return hasattr(option, "option_type") and self.supports(option.option_type)
+
+    def check_compatibility(self, options: list) -> tuple[bool, list[str]]:
+        """Check compatibility of options with this backend."""
+        if not options:
+            return True, []
+
+        for option in options:
+            if not self.supports_option(option):
+                unsupported = [
+                    f"{opt.__class__.__name__} ({opt.option_type.value})"
+                    for opt in options
+                    if not self.supports_option(opt)
+                ]
+                return False, unsupported
+
+        return True, []
+
+
+# Backend capability definitions
+KUBERNETES_CAPABILITIES = BackendCapabilities(
```

Comment:
Can you move this KUBERNETES_CAPABILITIES to `kubernetes/backend.py`, same as for container backend ?